service/worker/batcher/workflow.go

```diff
@@ -93,6 +93,12 @@ type (
 
 	// ResetParams is the parameters for reseting workflow
 	ResetParams struct {
+		// This is a serialized commonpb.ResetOptions. We can't include it with the
+		// correct type because workflow/activity arguments are going to be serialized with the
+		// json dataconverter, which doesn't support the "oneof" field in ResetOptions.
+		ResetOptions []byte
+		resetOptions *commonpb.ResetOptions // deserialized version
```

Comment:
@tdeebswihart This is the json + proto serialization problem I was talking about

Reply:
Ew. Yeah, the proto libraries won't work with non-proto structs and IIRC they make it impossible to manually implement the `proto.Message` interface